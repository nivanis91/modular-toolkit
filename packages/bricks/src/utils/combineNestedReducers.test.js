import { combineNestedReducers, createFakeActionAndReducer } from '.';
import { createStore } from 'redux';

const [foo, fooAction] = createFakeActionAndReducer('FOO');
const [bar, barAction] = createFakeActionAndReducer('BAR');
const [baz, bazAction] = createFakeActionAndReducer('BAZ');
const [bazong, bazongAction] = createFakeActionAndReducer('BAZONG');

describe('When I combine some nested reducers', () => {
    let reducer;
    beforeEach(() => {
        reducer = combineNestedReducers({
            foo,
            corge: { bar, xyzzy: { baz, bazong } }
        });
    });
    describe('and I create a store with the combined reducers', () => {
        let store;
        beforeEach(() => (store = createStore(reducer)));
        describe('the initial state', () => void it('is correct', () => expect(store.getState()).toMatchSnapshot()));
        describe('and I dispatch an action for each of the combined nested reducers', () => {
            beforeEach(() => {
                store.dispatch(fooAction);
                store.dispatch(barAction);
                store.dispatch(bazAction);
                store.dispatch(bazongAction);
            });
            describe('the resulting state', () =>
                void it('is correct', () => expect(store.getState()).toMatchSnapshot()));
        });
    });
});
