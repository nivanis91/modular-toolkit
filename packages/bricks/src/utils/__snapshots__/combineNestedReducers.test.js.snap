// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I combine some nested reducers and I create a store with the combined reducers and I dispatch an action for each of the combined nested reducers the resulting state is correct 1`] = `
Object {
  "corge": Object {
    "bar": Object {
      "value": "bar",
    },
    "xyzzy": Object {
      "baz": Object {
        "value": "baz",
      },
      "bazong": Object {
        "value": "bazong",
      },
    },
  },
  "foo": Object {
    "value": "foo",
  },
}
`;

exports[`When I combine some nested reducers and I create a store with the combined reducers the initial state is correct 1`] = `
Object {
  "corge": Object {
    "bar": Object {},
    "xyzzy": Object {
      "baz": Object {},
      "bazong": Object {},
    },
  },
  "foo": Object {},
}
`;
